spec/registry.xml

```diff
@@ -0,0 +1,250 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-model href="registry.dtd" type="application/xml-dtd"?>
+<registry>
+  <pattern id="anyNumber" regex="-?[0-9]+(\.[0-9]+)"/>
```

Comment:
Hard / impossible to represent properly with just regexp.
Created issue https://github.com/unicode-org/message-format-wg/issues/422 for this (and more)

Reply:
Here's a regexp matching the JSON number spec:
```suggestion
  <pattern id="anyNumber" regex="-?(0|([1-9]\d*))(\.\d*)([eE][-+]?\d+)"/>
```

Reply:
"0", // fails because there is no dot
"123", // fails because there is no dot
"+2", // fails because plus is not allowed
"3.14" // fails because it's missing the `eE...`, which is not optional

The regexp misses some `*` here and there.

---

But that's not the point.
The point is that in general regex are not strong enough to represent everything we might need.